pool.go

```diff
@@ -166,6 +178,70 @@ func (p *pool) refreshPool() {
 	}
 }
 
+func (p *pool) checkPool() {
+	for {
+		select {
+		case msg := <-p.mirrorSamples:
+			// see if it is to a main-tier node - if so find appropriate test node to test against.
+			p.lk.RLock()
+			if p.th.NodeTier(msg.node) != tieredhashing.TierMain {
+				p.lk.RUnlock()
+				continue
+			}
+			testNodes := p.th.GetNodes(tieredhashing.TierUnknown, msg.key, 1)
+			p.lk.RUnlock()
+			if len(testNodes) == 0 {
+				continue
+			}
+			trialTimeout, cancel := context.WithTimeout(context.Background(), time.Second*19)
+			_ = p.fetchResourceAndUpdate(trialTimeout, testNodes[0], msg.path, 0, p.mirrorValidator)
+			cancel()
+		case <-p.done:
+			return
+		}
+	}
+}
+
+func (p *pool) mirrorValidator(resource string, reader io.Reader) error {
```

Comment:
Please can we add a unit tests for `mirrorValidator ` happy and failure cases ?

Reply:
i'm tempted to not put too much work into this because we should get an actual validator in both boxo and lassie and should switch to one of those two once they exist